lib/Doctrine/Persistence/Mapping/AbstractClassMetadataFactory.php

```diff
@@ -400,6 +445,11 @@ public function getReflectionService()
         return $this->reflectionService;
     }
 
+    protected function getCacheKey(string $realClassName): string
+    {
+        return str_replace('\\', '__', $realClassName) . $this->cacheSalt;
```

Comment:
This (along with the changed cache salt) will invalidate existing caches. We'll have to check if existing implementations manually mess with cache keys, in which case we'll have to come back and check the impact of this change.